jerry-core/ecma/builtin-objects/ecma-builtin-array-iterator-prototype.c

```diff
@@ -81,25 +81,27 @@ ecma_builtin_array_iterator_prototype_object_next (ecma_value_t this_val) /**< t
     return ecma_create_iter_result_object (ECMA_VALUE_UNDEFINED, ECMA_VALUE_TRUE);
   }
 
-  uint32_t length;
-
   /* 8 - 9. */
-#ifndef CONFIG_DISABLE_ES2015_TYPEDARRAY_BUILTIN
-  if (ecma_is_typedarray (ecma_make_object_value (array_object_p)))
+  ecma_value_t len_value = ecma_op_object_get (array_object_p,
+                                               ecma_get_magic_string (LIT_MAGIC_STRING_LENGTH));
+
+  if (ECMA_IS_VALUE_ERROR (len_value))
   {
-    length = ecma_typedarray_get_length (array_object_p);
+    return len_value;
   }
-  else
-  {
-#endif /* !CONFIG_DISABLE_ES2015_TYPEDARRAY_BUILTIN */
-    JERRY_ASSERT (ecma_get_object_type (array_object_p) == ECMA_OBJECT_TYPE_ARRAY);
 
-    ecma_extended_object_t *ext_array_obj_p = (ecma_extended_object_t *) array_object_p;
+  ecma_number_t length_number;
+  ecma_value_t length_value = ecma_get_number (len_value, &length_number);
 
-    length = ext_array_obj_p->u.array.length;
-#ifndef CONFIG_DISABLE_ES2015_TYPEDARRAY_BUILTIN
+  if (!ecma_is_value_empty (length_value))
```

Comment:
`if (ECMA_IS_VALUE_ERROR (length_value))`

Reply:
Array iterator is an ES2015 feature so this change does not affect any es5.1 code path.